src/actions/ens/getEnsText.ts

```diff
@@ -20,21 +20,24 @@ import {
   readContract,
 } from '../public/readContract.js'
 
-export type GetEnsTextParameters = Prettify<
+export type GetEnsTextParameters<TKeys> = Prettify<
   Pick<ReadContractParameters, 'blockNumber' | 'blockTag'> & {
     /** ENS name to get Text for. */
     name: string
-    /** Text record to retrieve. */
-    key: string
+    /** Text record(s) to retrieve. */
+    key: TKeys
     /** Address of ENS Universal Resolver Contract. */
     universalResolverAddress?: Address
   }
 >
 
-export type GetEnsTextReturnType = string | null
+// if a single key is passed, return a single string or null. otherwise, return an array of strings or nulls.
+export type GetEnsTextReturnType<T> = T extends string
+  ? string | null
+  : (string | null)[] | null
```

Comment:
Removing conditional type:

```suggestion
export type GetEnsTextReturnType<TKeys extends  string | readonly string[] = string | readonly string[]> =
  | (TKeys extends string ? string | null : never)
  | (TKeys extends readonly string[] ? (string | null)[] | null : never)
```